src/components/track/desktop/stats/StatsText.tsx

```diff
@@ -118,9 +116,7 @@ const StatsText = memo(
 
 function mapStateToProps(state: AppState, ownProps: OwnProps) {
   return {
-    user1: getUser(state, { id: ownProps.userId1 }),
-    user2: getUser(state, { id: ownProps.userId2 }),
-    user3: getUser(state, { id: ownProps.userId3 })
+    users: Object.values(getUsers(state, { ids: ownProps.userIds }))
```

Comment:
this was actually on purpose. we should play with the performance a bit, but the reason was that

`getUser` will return the exact cached entry (https://github.com/AudiusProject/audius-client/blob/f36aaf2dd555a6269edc7cca81b87935407e75ce/src/common/store/cache/users/selectors.ts#L15) that is the same object ref (https://github.com/AudiusProject/audius-client/blob/f36aaf2dd555a6269edc7cca81b87935407e75ce/src/common/store/cache/selectors.ts#L60) each time. Only when that individual user in the cache gets replaced, we would rerender here

versus `getUsers` which will return a *new* object each time: https://github.com/AudiusProject/audius-client/blob/f36aaf2dd555a6269edc7cca81b87935407e75ce/src/common/store/cache/users/selectors.ts#L43, which will cause a rerender every time a user is added to the cache (unfortunate).

Ideally we would connect this component all the way down to each user level to prevent this.

Reply:
Oof good callout - could we avoid rerenders by calling `ownProps.userIds.map(id => getUser(state, { id }))`

Reply:
wouldn't the `.map` produce a new list each time as well?

Reply:
i think you'd need to use reselect / similar to control the comparison

Reply:
yeah that's what I was guessing

Reply:
https://github.com/AudiusProject/audius-client/pull/657/commits/f8c55e44dbadb19b4b6650d4373a24363df5517b Should that do it?

Reply:
My guess is no, because it still depends on all of `getUsers` but maybe I'm wrong?

Reply:
Updated to use createShallowSelector and verified it reduced rerenders